packages/component-elife-ui/client/molecules/ModalOverlay.test.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { render, cleanup } from '@testing-library/react'
+import theme from '@elifesciences/elife-theme'
+import { ThemeProvider } from 'styled-components'
+import { MockedProvider } from 'react-apollo/test-utils'
+import 'jest-dom/extend-expect'
+import ModalOverlay from './ModalOverlay'
+
+theme.transitionDuration = '0.2s'
+
+const setupProvider = ({ children }) => (
+  <ThemeProvider theme={theme}>
+    <MockedProvider>{children}</MockedProvider>
+  </ThemeProvider>
+)
+
+describe('ModalOverlay', () => {
+  it('should disable scrolling on body element when open', () => {
+    const { baseElement, rerender } = render(
+      <ModalOverlay open={false} transparentBackground>
+        Modal Content
+      </ModalOverlay>,
+      { wrapper: setupProvider },
+    )
+    rerender(
+      <ModalOverlay open transparentBackground>
+        Modal Content
+      </ModalOverlay>,
+      { wrapper: setupProvider },
+    )
+
+    expect(baseElement).toHaveStyle(`overflow: hidden`)
+
+    cleanup()
+  })
+
+  it('should enable scrolling on body element when closed', () => {
+    const { baseElement } = render(
+      <ModalOverlay open={false} transparentBackground>
+        Modal Content
+      </ModalOverlay>,
+      { wrapper: setupProvider },
+    )
+
+    expect(baseElement).not.toHaveStyle(`overflow: hidden`)
+
+    cleanup()
```

Comment:
Could be put in an `afterEach()` ?

Reply:
Done and also refactored component render inside factory function